### ISSUE TITLE
Don't save files as ascii in the editor if there's no coding line

### ISSUE THREAD

**Issue by ccordoba12** [spyder-ide/spyder#12416]
Right now we have the following problems with these operations, that could be causing issues with our LSP architecture:

* Files that don't have the initial `coding` line (i.e. `# -*- coding: utf-8 -*-`) are saved as ascii. Instead, we should be save them as unicode, as VSCode does.

    I've seen that a lot of people remove all contents from our template before starting to code, which causes this issue.

* Our process to load files from the previous session or after opening a project is very convoluted (it seems we're opening and closing those files several times), and that can confuse LSP servers a lot. Instead we should simply open files once.

    This problem could have been introduced in PR https://github.com/spyder-ide/spyder/pull/3858

@andfoy, please take a look at this one.

**Comment by goanpeca** [comment]
We could disable the repaints on the widget so that it remains "static" while other files load? Also, opening files on startup has a noticeable lag.

Opening 10 o 20 files should not feel stucked on every file. If we are doing some GUI blocking stuff on each file on open we should defer them until the file is focused.

**Comment by ccordoba12** [comment]
> Opening 10 o 20 files should not feel stucked on every file. If we are doing some GUI blocking stuff on each file on open we should defer them until the file is focused.

Please open a new issue about this. @andfoy and me discussed some ideas on how to improve this.
